examples/platform/linux/AppMain.cpp

```diff
@@ -101,6 +101,7 @@ using namespace chip::Inet;
 using namespace chip::Transport;
 using namespace chip::app::Clusters;
 
+#if EMBER_AF_PLUGIN_NETWORK_COMMISSIONING_SERVER
```

Comment:
I don't think this file includes any headers that would define that macro....

Has his been tested with a WiFi or Thread device running Linux, @erwinpan1 ?

Reply:
The file is not included. Thanks for catching this. 
I'll need to find another approach.

Reply:
More generally, AppMain is not dependent on the ZAP configuration....

Reply:
Does this make sense? ZAP actually determines it's capability and the initialization or AppMain, right?

Reply:
Well, concretely when examples/platform/linux/BUILD.gn compiles things it does not depend on any ZAP-dependent bits....